graylog2-server/src/main/java/org/graylog2/plugin/utilities/date/DateTimeConverter.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog2.plugin.utilities.date;
+
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+
+import javax.annotation.Nonnull;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.Date;
+import java.util.TimeZone;
+
+import static org.graylog2.plugin.Tools.ES_DATE_FORMAT_FORMATTER;
+
+public class DateTimeConverter {
+    public static DateTime convertToDateTime(@Nonnull Object value) {
+        if (value instanceof DateTime) {
+            return (DateTime) value;
+        }
+
+        if (value instanceof Date) {
+            return new DateTime(value, DateTimeZone.UTC);
+        } else if (value instanceof ZonedDateTime) {
+            final DateTimeZone dateTimeZone = DateTimeZone.forTimeZone(TimeZone.getTimeZone(((ZonedDateTime) value).getZone()));
+            return new DateTime(Date.from(((ZonedDateTime) value).toInstant()), dateTimeZone);
```

Comment:
Is there a reason why for `ZonedDateTime` we now try to preserve the time offset while for `Date`, `OffsetDateTime` and `Instant` we are now removing that information and reverting to UTC?

Reply:
I think it's best to preserve as much as possible.
The actual conversion to UTC will be performed when it's written to elastic

Reply:
I agree that we should preserve as much as possible, but unless I'm wrong, with your change we are now removing offset information where it was previously preserved in the `Date`, `OffsetDateTime` and `Instant` case. So to rephrase: let's be consistent by either preserving as much as possible all the time or converting everything to UTC.

Reply:
Ok, so we are only "losing" the system default offset which is not worth keeping. I think the solution is good as it is 👍